diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5df26e5..02d07db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,13 +72,13 @@ macro_log_feature(ZLIB_FOUND "ZLib" "Support for gzip compressed files and data
 
 set(STRIGI_MIN_VERSION 0.6.3)
 find_package(Strigi)
-if (WIN32)
+if (WIN32 OR APPLE)
     set (STRIGI_REQUIRED FALSE)
     set (STRIGI_EXTRA_TEXT "")
-else (WIN32)
+else (WIN32 OR APPLE)
     set (STRIGI_REQUIRED TRUE)
     set (STRIGI_EXTRA_TEXT "Required by some critical kioslaves")
-endif (WIN32)
+endif (WIN32 OR APPLE)
 macro_log_feature(STRIGI_FOUND "Strigi" "Desktop indexing and search support" "http://strigi.sourceforge.net" ${STRIGI_REQUIRED} "${STRIGI_MIN_VERSION}" "${STRIGI_EXTRA_TEXT}")
 
 set(LIBATTICA_MIN_VERSION "0.1.90")
diff --git a/kio/kio/kfilemetainfo.h b/kio/kio/kfilemetainfo.h
index 6920ffe..4fc6eae 100644
--- a/kio/kio/kfilemetainfo.h
+++ b/kio/kio/kfilemetainfo.h
@@ -117,7 +117,7 @@ public:
     /**
      * @brief Copy a KFileMetaInfo instance from another one.
      **/
-    KFileMetaInfo& operator=(KFileMetaInfo const& kfmi);
+    const KFileMetaInfo& operator=(KFileMetaInfo const& kfmi);
     /**
      * @brief Save the changes made to this KFileMetaInfo instance.
      */
